.github/workflows/build-and-deploy-on-pypi.yml

```diff
@@ -0,0 +1,47 @@
+name: PyPi Build and Deploy 🐍📦
+
+on:
+  release:
+    types: [published]
+  # use this for testing
+  push:
+    branches:
+      - main
+
+jobs:
+  build-n-publish:
+    name: Build and publish MyProxyClient on PyPi
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Set up Python 3.12
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.12"
+      - name: Install pep517
+        run: >-
+          python -m
+          pip install
+          pep517
+          --user
+      - name: Build a binary wheel and a source tarball
+        run: >-
+          python -m
+          pep517.build
+          --source
+          --binary
+          --out-dir dist/
+          .
+      #- name: Publish distribution 📦 to Test PyPI
+      #  uses: pypa/gh-action-pypi-publish@master
+      #  with:
+      #    password: ${{ secrets.test_pypi_password }}
+      #    repository_url: https://test.pypi.org/legacy/
+      - name: Publish distribution 📦 to PyPI
+        if: startsWith(github.ref, 'refs/tags')
+        uses: pypa/gh-action-pypi-publish@release/v1
+        with:
+          user: __token__
```

Comment:
You may want to have a look at trusted publishers instead of using a username and password, is the new thing: https://blog.pypi.org/posts/2023-04-20-introducing-trusted-publishers/

Reply:
ah evolution! Great pointer @bouweandela - cheers! Do I take it I can ask you for reviews in the future? Feels very lonely around 'ere 😁

Reply:
Yes, I would be happy to help out :beers: